torchani/units.py

```diff
@@ -0,0 +1,140 @@
+EV_TO_KJOULEMOL = EV_TO_JOULE * AVOGADROS_NUMBER / 1000
+
+# For vibrational analysis:
+
+INVCM_TO_EV = 0.0001239841973964072  # equal to ase.units.invcm
+# To convert from the sqrt of eigenvalues (mass-scaled hessian units of
+# sqrt(hartree / (amu * A^2)) into cm^-1
+# it is necessary to multiply by the sqrt ( HARTREE_TO_JOULE / AMU_TO_KG ),
+# then we convert angstroms to meters and divide by 1/ speed_of_light (to
+# convert seconds into meters). Finally divide by 100 to get inverse cm
+# The resulting value should be close to 17092
+SQRT_MHESSIAN_TO_INVCM = (math.sqrt(HARTREE_TO_JOULE / AMU_TO_KG) / ANGSTROM_TO_METER / SPEED_OF_LIGHT) / 100
+# meV is other common unit used to present vibrational transition energies
+SQRT_MHESSIAN_TO_MILLIEV = SQRT_MHESSIAN_TO_INVCM * INVCM_TO_EV * 1000
+# To convert units form mass-scaled hessian units into mDyne / Angstrom (force
+# constant units) factor should be close to 4.36
+MHESSIAN_TO_FCONST = HARTREE_TO_JOULE * NEWTON_TO_MILLIDYNE / ANGSTROM_TO_METER
+
+
+def sqrt_mhessian2invcm(x):
+    r"""Converts sqrt(mass-scaled hessian units) into cm^-1
+
+    Converts form units of sqrt(Hartree / (amu * Angstrom^2))
+    which are sqrt(units of the mass-scaled hessian matrix)
+    into units of inverse centimeters.
+
+    Take into account that to convert the actual eigenvalues of the hessian
+    into wavenumbers it is necessary to multiply by an extra factor of 1 / (2 *
+    pi)"""
+    return x * SQRT_MHESSIAN_TO_INVCM
+
+
+def sqrt_mhessian2milliev(x):
+    r"""Converts sqrt(mass-scaled hessian units) into meV
+
+    Converts form units of sqrt(Hartree / (amu * Angstrom^2))
+    which are sqrt(units of the mass-scaled hessian matrix)
+    into units of milli-electronvolts.
+
+    Take into account that to convert the actual eigenvalues of the hessian
+    into wavenumbers it is necessary to multiply by an extra factor of 1 / (2 *
+    pi)"""
+    return x * SQRT_MHESSIAN_TO_MILLIEV
+
+
+def mhessian2fconst(x):
+    r"""Converts mass-scaled hessian units into mDyne/Angstrom
+
+    Converts from units of mass-scaled hessian (Hartree / (amu * Angstrom^2)
+    into force constant units (mDyne/Angstom), where 1 N = 1 * 10^8 mDyne"""
+    return x * MHESSIAN_TO_FCONST
+
+
+def hartree2ev(x):
+    r"""Hartree to eV conversion factor from 2014 CODATA"""
+    return x * HARTREE_TO_EV
+
+
+def ev2kjoulemol(x):
+    r"""Electronvolt to kJ/mol conversion factor from CODATA 2014"""
+    return x * EV_TO_KJOULEMOL
+
+
+def ev2kcalmol(x):
+    r"""Electronvolt to kcal/mol conversion factor from CODATA 2014"""
+    return x * EV_TO_KCALMOL
+
+
+def hartree2kjoulemol(x):
+    r"""Hartree to kJ/mol conversion factor from CODATA 2014"""
+    return x * HARTREE_TO_KJOULEMOL
+
+
+def hartree2kcalmol(x):
+    r"""Hartree to kJ/mol conversion factor from CODATA 2014"""
+    return x * HARTREE_TO_KCALMOL
+
+
+# Add actual values to docstrings on import
+hartree2ev.__doc__ += f'\n\n1 Hartree = {hartree2ev(1)} eV'
```

Comment:
the type checking in mypy doesn't like me doing this, is this bad?

Reply:
Maybe you should add something like `assert hartree2ev.__doc__ is not None`, this error happens because the type checker thinks `hartree2ev.__doc__` could be `None`, but `None` does not have a `+=`.

Adding the assert might allow the type checker to do type narrowing, so the type of `hartree2ev.__doc__` can now be inferred as `str` instead of `Optional[str]`.

I am not very sure if mypy can correctly do type narrowing the way I described (torch.jit does so), but it worth a try. But if adding the `assert` does not work, then you could try something like
`hartree2ev.__doc__ = '' if hartree2ev.__doc__ is None` or anything that you think could help the type system to infer type better.

Reply:
Ahhh, that is very interesting. I think what I ended up doing would convert ``None`` to ``''`` so that is why it worked. It was strange to me that it worked, now I get it.